src/main/java/br/com/zup/pagamentos/transacao/Transacao.java

```diff
@@ -0,0 +1,90 @@
+package br.com.zup.pagamentos.transacao;
+
+import br.com.zup.pagamentos.formapagamento.FormaPagamento;
+import br.com.zup.pagamentos.restaurante.Restaurante;
+import br.com.zup.pagamentos.usuario.Usuario;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+import static javax.persistence.EnumType.STRING;
+
+@Entity
+public class Transacao {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
```

Comment:
vc está usando o `@Column(nullable = false)` como dica pro dev? 

pergunto isso pois vc está usando migrations então isso não eh necessário e pode até atrapalhar (confundir) se vc nao manter ambos os modelos sincronizados: entidade e schema do tabela. 

se for pra dar uma dica para o dev, entao eh menor usar a `@NotNull` na qual não só dá a dica como garante constraints a nível de aplicação. Dá uma lida depois aqui, tem coisas interessantes: https://www.baeldung.com/hibernate-notnull-vs-nullable

Reply:
foi pra dica e também pra me ajudar com o diff do migrations haha.